### PR TITLE
Changes for common typings

### PR DESCRIPTION
**Pull request by hanzlamateen**
## Summary

_A summary of changes being made in this PR_

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

_References to pertaining issue(s)_

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

_Reviewers for this PR_

### CODE REVIEW

**Review by kimenyikevin — COMMENTED** [review]
It looks good to me, However we can start avoiding to use ``any`` in Exrengine.

File: packages/server-core/src/user/identity-provider/identity-provider.class.ts [hunk 1]
```diff
@@ -38,7 +41,7 @@ export class IdentityProvider extends Service {
    * @param params
    * @returns accessToken
    */
-  async create(data: any, params: Params): Promise<any> {
+  async create(data: any, params: Params = {}): Promise<T & { accessToken?: string }> {
```
Comment (kimenyikevin) [inline packages/server-core/src/user/identity-provider/identity-provider.class.ts:44]: Like ``data:any`` can be changed

Comment (HexaField) [reply packages/server-core/src/user/identity-provider/identity-provider.class.ts:44]: yes, sorry i meant to assign you not request a review. I will provide some more details on what is needed for this PR soon.

Comment (kimenyikevin) [reply packages/server-core/src/user/identity-provider/identity-provider.class.ts:44]: Great

**Pull request merged**